zip-0227.rst

```diff
@@ -9,7 +9,614 @@
+
+Specification: Global Issuance State
+====================================
+
+Issuance requires the following additions to the global state defined at block boundaries:
+
+- :math:`\mathsf{previously\_finalized}`, a set of :math:`\mathsf{AssetId}` that have been finalized (i.e.: the :math:`\mathsf{finalize}` flag has been set to :math:`1` in some issuance transaction preceding the block boundary).
+
+
+Specification: Issuance Action, Issuance Bundle and Issuance Protocol
+=====================================================================
+
+Issuance Action Description
+---------------------------
+
+An issuance action, ``IssueAction``, is the instance of issuing a specific Custom Asset, and contains the following fields:
+
+- :math:`\mathsf{assetDescSize}`: the size of the Asset description, a number between :math:`0` and :math:`512`, stored in two bytes.
+- :math:`\mathsf{asset\_desc}`: the Asset description, a byte string of up to 512 bytes as defined in the `Specification: Asset Identifier`_ section.
+- ``vNotes``: an array of ``Note`` containing the unencrypted output notes of the recipients of the Asset.
+- ``flagsIssuance``: a byte that stores the :math:`\mathsf{finalize}` boolean that defines whether the issuance of that specific Custom Asset is finalized or not.
+
+An asset's :math:`\mathsf{AssetDigest}` is added to the :math:`\mathsf{previously\_finalized}` set after a block that contains any issuance transaction for that asset with :math:`\mathsf{finalize} = 1`. It then cannot be removed from this set. For Assets with :math:`\mathsf{AssetDigest} \in \mathsf{previously\_finalized}`, no further tokens can be issued, so as seen below, the validators will reject the transaction. For Assets with :math:`\mathsf{AssetDigest} \not\in \mathsf{previously\_finalized}`, new issuance actions can be issued in future transactions. These must use the same Asset description, :math:`\mathsf{asset\_desc}`, and can either maintain :math:`\mathsf{finalize} = 0` or change it to :math:`\mathsf{finalize} = 1`, denoting that this Custom Asset cannot be issued after the containing block.
+
+
++-----------------------------+--------------------------+-------------------------------------------+---------------------------------------------------------------------+
+| Bytes                       | Name                     | Data Type                                 | Description                                                         |
++=============================+==========================+===========================================+=====================================================================+
+|``2``                        |``assetDescSize``         |``byte``                                   |The length of the :math:`\mathsf{asset\_desc}` string in bytes.      |
```

Comment:
```suggestion
|``2``                        |``assetDescSize``         |``uint16``                                 |The length of the :math:`\mathsf{asset\_desc}` string in bytes.      |
```
or use CompactSize. This table is duplicated between ZIPs 227 and 230; I'm not sure whether that's a problem but it should be called out if it stays like that.

Reply:
Removed the duplication, included it solely in ZIP 230.